File: src/main/java/com/businessrank/debttracker/model/Payment.java
package com.businessrank.debttracker.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Debt is required")
    @ManyToOne
    @JoinColumn(name = "debt_id")
    private Debt debt;

    @NotNull(message = "Amount is required")
    @DecimalMin(value = "0.01", message = "Amount must be greater than 0")
    @Column(precision = 10, scale = 2)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    private PaymentMethod paymentMethod;

    private String notes;

    private LocalDateTime paymentDate;

    private LocalDateTime createdAt;

    public enum PaymentMethod {
        CASH, CARD
    }

    // Constructors
    public Payment() {}

    public Payment(Debt debt, BigDecimal amount, PaymentMethod paymentMethod, String notes) {
        this.debt = debt;
        this.amount = amount;
        this.paymentMethod = paymentMethod;
        this.notes = notes;
        this.paymentDate = LocalDateTime.now();
        this.createdAt = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Debt getDebt() { return debt; }
    public void setDebt(Debt debt) { this.debt = debt; }

    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }

    public PaymentMethod getPaymentMethod() { return paymentMethod; }
    public void setPaymentMethod(PaymentMethod paymentMethod) { this.paymentMethod = paymentMethod; }

    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }

    public LocalDateTime getPaymentDate() { return paymentDate; }
    public void setPaymentDate(LocalDateTime paymentDate) { this.paymentDate = paymentDate; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: src/services/payment.ts
import api from './api';

export interface Payment {
  id: number;
  debt: {
    id: number;
    description?: string;
  };
  amount: number;
  paymentMethod: 'CASH' | 'CARD';
  notes?: string;
  paymentDate: string;
  createdAt: string;
}

export interface AddPaymentRequest {
  debtId: number;
  amount: number;
  paymentMethod: 'CASH' | 'CARD';
  notes?: string;
}

export interface UpdatePaymentRequest {
  amount: number;
  paymentMethod: 'CASH' | 'CARD';
  notes?: string;
}

export const paymentService = {
  getAllPayments: async (): Promise<Payment[]> => {
    const response = await api.get('/api/payments');
    return response.data;
  },

  getPaymentsByDebt: async (debtId: number): Promise<Payment[]> => {
    const response = await api.get(`/api/payments/debt/${debtId}`);
    return response.data;
  },

  getPaymentById: async (id: number): Promise<Payment> => {
    const response = await api.get(`/api/payments/${id}`);
    return response.data;
  },

  addPayment: async (paymentData: AddPaymentRequest): Promise<{ message: string; payment: Payment }> => {
    const response = await api.post('/api/payments', paymentData);
    return response.data;
  },

  updatePayment: async (id: number, paymentData: UpdatePaymentRequest): Promise<Payment> => {
    const response = await api.put(`/api/payments/${id}`, paymentData);
    return response.data;
  },

  deletePayment: async (id: number): Promise<{ message: string }> => {
    const response = await api.delete(`/api/payments/${id}`);
    return response.data;
  }
};